core/src/test/java/com/linecorp/armeria/server/ServerBuilderTest.java

```diff
@@ -20,16 +20,49 @@
 
 import java.util.List;
 
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.InMemoryDnsResolver;
```

Comment:
Could you use Armeria's `HttpClient`? Please take a look into `HttpClientSniTest.init()` that uses `ClientFactoryBuilder.addressResolverGroupFactory()` with `MockAddressResolverGroup`.